src/main/scala/org/scalameter/Executor.scala

```diff
@@ -30,21 +32,22 @@ object Executor {
   import Key._
 
   type Warmer = org.scalameter.Warmer
-  type Measurer = org.scalameter.Measurer
+  type Measurer[T] = org.scalameter.Measurer[T]
 
   val Warmer = org.scalameter.Warmer
   val Measurer = org.scalameter.Measurer
 
-  object None extends Executor {
-    def measurer: Measurer = ???
+  object None extends Executor[Nothing] {
+    def measurer: Measurer[Nothing] = ???
 
     def warmer: Warmer = ???
 
-    def runSetup[T](setup: Setup[T]): CurveData = ???
+    def runSetup[T](setup: Setup[T]): CurveData[Nothing] = ???
   }
 
-  trait Factory[E <: Executor] {
-    def apply(warmer: Warmer, aggregator: Aggregator, m: Measurer): E
+  trait Factory[E[_] <: Executor[_]] {
+    def apply[T: Numeric: Pickler](warmer: Warmer,
```

Comment:
I agree about the `Pickler` then, but I still think the case for `Numeric` is not strong enough, particularly because that type parameter will not always refer to a numeric type.

It would be better to introduce your own `PrettyPrinter` typeclass, which is derived directly from numeric for numeric types:

    implicit def numericPrettyPrinter[T: Numeric] = ...
    implicit def defaultPrettyPrinter[T] = ...